### ISSUE TITLE
hasNext() moves current position

### ISSUE THREAD

**Issue by dsukhoroslov** [pgelinas/jackson-javax-json#5]
I'm parsing JSON this way:

```
    while (parser.hasNext()) {
        processEvent(parser.next());
    }
```

the first event it process is KEY_NAME instead of initial START_OBJECT.

**Comment by pgelinas** [comment]
Allright, I'll try to look at this soon.

**Comment by dvilaverde** [comment]
The version of this library at http://central.maven.org/maven2/com/github/pgelinas/jackson-javax-json/0.9.0/ seems to have the hasNext() problem because the java class JacksonParser.java doesn't seem to have any of the fixes from commit https://github.com/pgelinas/jackson-javax-json/commit/c3210545b4c8277dcf3d0a4bb3bb28c60d981a99

If you notice that commit introduced a fix 

```
public Event next() {
         if (!_hasNextCalled) {
             advanceParser();
         }
         _hasNextCalled = false;
```

but the decompiled code up on maven central has this

```
public JsonParser.Event next()
  {
    advanceParser();
    JsonParser.Event event = (JsonParser.Event)_tokenToEvent.get(this._nextToken);
    if (event == null) {
      throw new NoSuchElementException();
    }
    return event;
  }
```

missing the`if (!_hasNextCalled) {` line around the advanceParser() call.

Can we get a 0.9.1 version put up on Maven Central?

**Comment by jean-morissette** [comment]
We also need this fix. Are you still supporting this project?

**Comment by pgelinas** [comment]
Should be fixed with 1.0.0 just released. Sorry for the delay.
